cmd/applyconfig/applyconfig.go

```diff
@@ -22,22 +24,31 @@ import (
 )
 
 type command string
+type dryRunMethod string
 
 type options struct {
-	confirm     bool
 	user        *nullableStringFlag
 	directories flagutil.Strings
 	context     string
 	kubeConfig  string
+	dryRun      dryRunMethod
 }
 
 const (
 	ocApply   command = "apply"
 	ocProcess command = "process"
+	ocVersion command = "version"
+
+	dryNone   dryRunMethod = ""
+	dryAuto   dryRunMethod = "auto"
```

Comment:
Ah I did indeed miss your detection code. But then follow-up: Why would we ever not want to use `auto`?

Reply:
I think we'll want `auto` most of the time. My plan was to make it default *after* I fix the validation problems in our manifests (using a `--dry-run-method=auto` rehearsal - https://github.com/openshift/release/pull/14693) -- I assumed they can be fixed, not sure about that now seeing your other comment.

I generally like to have flags of this kind, they allow flexibility when needed (dunno, `auto` can guess wrong and we may need to keep PRs going, that kind of thing). I don't think it's that much of a burden, `auto` has 99% of the complexity anyway.

Reply:
They can be fixed, I think all of these issues is ppl using `v1` as the apiGroup instead of the real apigroup for resources served by the openshift apiserver. For compatibility reasons `oc` translates that, but it breaks the openapi validation.

Reply:
Which is deprecated, but oc users do not see that so no one has an incentive to stop cargoculting it from fold manifests: https://coreos.slack.com/archives/CB48XQ4KZ/p1610042971090100